frontend/tests/pages/maintenance/page.test.tsx

```diff
@@ -20,22 +21,146 @@ jest.mock("react", () => ({
   })),
 }));
 
+const redirectMock = jest.fn();
+
+jest.mock("next/navigation", () => ({
+  useRouter() {
+    return {
+      prefetch: () => null,
+    };
+  },
+  usePathname() {
+    return {
+      prefetch: () => null,
+    };
+  },
+  push: jest.fn(),
+  redirect: (url: string): unknown => redirectMock(url),
+  RedirectType: { push: "PUSH" },
+}));
+
 jest.mock("next-intl", () => ({
   useTranslations: () => useTranslationsMock(),
   useMessages: () => mockMessages,
 }));
 
 describe("Maintenance", () => {
   it("renders intro text", () => {
-    render(<Maintenance params={localeParams} />);
-
+    // set a major feature offline so we don't redirect to homepage
+    render(
+      <UserContext
```

Comment:
this is a lot of mocking to do - worth looking into the pattern used in the `FakeQueryProvider` in case we want to make this more reusable